File: VB/PrismOnDXDocking.Infrastructure/Adapters/DockManagerAdapter.vb
Imports System.ComponentModel.Composition
Imports System.Windows.Controls
Imports DevExpress.Xpf.Docking
Imports Prism.Regions

Namespace PrismOnDXDocking.Infrastructure.Adapters

    <Export(GetType(DockManagerAdapter)), PartCreationPolicy(CreationPolicy.NonShared)>
    Public Class DockManagerAdapter
        Inherits RegionAdapterBase(Of DockLayoutManager)

        <ImportingConstructor>
        Public Sub New(ByVal behaviorFactory As IRegionBehaviorFactory)
            MyBase.New(behaviorFactory)
        End Sub

        Protected Overrides Function CreateRegion() As IRegion
            Return New SingleActiveRegion()
        End Function

        Protected Overrides Sub Adapt(ByVal region As IRegion, ByVal regionTarget As DockLayoutManager)
            Dim items As BaseLayoutItem() = regionTarget.GetItems()
            For Each item As BaseLayoutItem In items
                Dim regionName As String = RegionManager.GetRegionName(item)
                If Not String.IsNullOrEmpty(regionName) Then
                    Dim panel As LayoutPanel = TryCast(item, LayoutPanel)
                    If panel IsNot Nothing AndAlso panel.Content Is Nothing Then
                        Dim control As ContentControl = New ContentControl()
                        RegionManager.SetRegionName(control, regionName)
                        panel.Content = control
                    End If
                End If
            Next
        End Sub
    End Class
End Namespace

File: CS/PrismOnDXDocking.Infrastructure/Adapters/DockManagerAdapter.cs
using System.ComponentModel.Composition;
using System.Windows.Controls;
using DevExpress.Xpf.Docking;
using System;
using Prism.Regions;

namespace PrismOnDXDocking.Infrastructure.Adapters {
    [Export(typeof(DockManagerAdapter)), PartCreationPolicy(CreationPolicy.NonShared)]
	public class DockManagerAdapter : RegionAdapterBase<DockLayoutManager> {
        [ImportingConstructor]
		public DockManagerAdapter(IRegionBehaviorFactory behaviorFactory) : 
            base(behaviorFactory) {
		}
		protected override IRegion CreateRegion() {
			return new SingleActiveRegion();
		}
		protected override void Adapt(IRegion region, DockLayoutManager regionTarget) {
			BaseLayoutItem[] items = regionTarget.GetItems();
            foreach(BaseLayoutItem item in items) {
                string regionName = RegionManager.GetRegionName(item);
                if(!String.IsNullOrEmpty(regionName)) {
                    LayoutPanel panel = item as LayoutPanel;
                    if(panel != null && panel.Content == null) {
                        ContentControl control = new ContentControl();
                        RegionManager.SetRegionName(control, regionName);
                        panel.Content = control;
                    }
                }
            }
		}
	}
}